### PR TITLE
Add hardware details filters

### LINKED ISSUE #572: implement frontend filters in hardware details
Part of #447

### PR DESCRIPTION
**Pull request by Francisco2002**
In this PR, I add the filters in hardware details in frontend and add counts in hardware tabs.

Close #572

### CODE REVIEW

File: dashboard/src/pages/hardwareDetails/Tabs/Build/BuildTab.tsx [hunk 1]
```diff
@@ -1,33 +1,270 @@
...
+//TODO: put it in other file to be reused
+export const MemoizedStatusCard = memo(StatusCard);
+
+const SummaryItem = ({
+  arch,
+  compilers,
+  onClickKey,
+  leftIcon,
+}: ISummaryItem): JSX.Element => {
+  //const diffFilter = useDiffFilterParams(arch.text, 'archs');
```
Comment (WilsonNet) [inline dashboard/src/pages/hardwareDetails/Tabs/Build/BuildTab.tsx:112]: leftover?

File: dashboard/src/types/hardware/hardwareDetails.ts [hunk 2]
```diff
@@ -94,7 +95,75 @@ export const zDiffFilter = z
...
+
+export const isTFilterObjectKeys = (key: string): key is TFilterObjectsKeys => {
+  return zFilterObjectsKeys.safeParse(key).success;
+};
+
+export const isTFilterNumberKeys = (key: string): key is TFilterNumberKeys => {
+  return zFilterNumberKeys.safeParse(key).success;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (WilsonNet) [inline dashboard/src/types/hardware/hardwareDetails.ts:116]: add TODO to remove this

File: dashboard/src/types/hardware/hardwareDetails.ts [hunk 3]
```diff
@@ -94,7 +95,75 @@ export const zDiffFilter = z
...
+const requestFilters = {
```
Comment (lfjnascimento) [inline dashboard/src/types/hardware/hardwareDetails.ts:117]:
```suggestion

type requestFilters = {
```

File: dashboard/src/types/hardware/hardwareDetails.ts (hunk 2 above)
Comment (lfjnascimento) [reply dashboard/src/types/hardware/hardwareDetails.ts:116]: just change the declartion to a type
```suggestion
type requestFilters = {
```

File: dashboard/src/pages/hardwareDetails/Tabs/Build/BuildTab.tsx [hunk 4]
```diff
@@ -1,33 +1,270 @@
...
+import { BuildStatus, GroupedTestStatus } from '@/components/Status/Status';
+import { DumbListingContent } from '@/components/ListingContent/ListingContent';
+
+import ListingItem from '@/components/ListingItem/ListingItem';
+
+import { TableCell, TableCellWithLink, TableRow } from '@/components/ui/table';
+
+import FilterLink from '../../HardwareDetailsFilterLink';
+
+//import { useDiffFilterParams } from '../../hardwareDetailsUtils';
```
Comment (lfjnascimento) [inline dashboard/src/pages/hardwareDetails/Tabs/Build/BuildTab.tsx:35]: remove it

**Comment by lfjnascimento** [comment]
Create a TODO to remind us to review the need for those file duplications

**Review by WilsonNet — APPROVED** [review]
very nice job, please add some TODOs in the files that you copied so when you continue the filters you make components of those

**Review by lfjnascimento — APPROVED** [review]
It's working, good job

File: dashboard/src/types/hardware/hardwareDetails.ts (hunk 2 above)
Comment (Francisco2002) [reply dashboard/src/types/hardware/hardwareDetails.ts:116]: I use `const targetFilter: readonly string[] = requestFilters[target];` in a function below. Change for type give a error in typescript

Comment (lfjnascimento) [reply dashboard/src/types/hardware/hardwareDetails.ts:117]: If you are using it you don't need the eslint-disable

**Pull request merged**